### PR TITLE
Install mlpack and cereal headers in R package

### PR DESCRIPTION
**Pull request by rcurtin**
This handles #3486.

By putting the `mlpack` and `cereal` headers into `inst/include/`, they will be installed as part of the R package installation process.  I also had to update `Makevars` to correctly include the headers (in the same way that's done in the `Makevars` of `RcppEnsmallen`).

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Just got in from 25 miles on the bicycle where I realized ... we are doing this likely wrong :)  When you install the headers into `src/`, they should go directly to `inst/include/` and `src/Makevars` should get a `-I../inst/include` to use them during package build.  Anyway...  Will review now.

File: src/mlpack/bindings/R/CMakeLists.txt [hunk 1]
```diff
@@ -418,7 +418,7 @@ macro (post_r_setup)
   # First we have to create that directory though.
   execute_process(
     COMMAND ${CMAKE_COMMAND} -E
-      make_directory ${CMAKE_BINARY_DIR}/src/mlpack/bindings/R/mlpack/src/mlpack
+      make_directory ${CMAKE_BINARY_DIR}/src/mlpack/bindings/R/mlpack/inst/include/mlpack
```
Comment (eddelbuettel) [inline src/mlpack/bindings/R/CMakeLists.txt:421]: Exactly!  You noticed that too. So disregard the comment I just made :)

File: src/mlpack/bindings/R/mlpack/src/Makevars [hunk 2]
```diff
@@ -1,2 +1,2 @@
-PKG_CXXFLAGS = -DBOOST_MATH_PROMOTE_DOUBLE_POLICY=false -I. $(SHLIB_OPENMP_CXXFLAGS)
+PKG_CXXFLAGS = -I. -I../inst/include $(SHLIB_OPENMP_CXXFLAGS)
```
Comment (eddelbuettel) [inline src/mlpack/bindings/R/mlpack/src/Makevars:1]: Spot on again.

**Review by eddelbuettel — APPROVED** [review]
Excellent stuff -- :ship:

**Review by zoq — APPROVED** [review]
Looks good to me.

**Comment by rcurtin** [comment]
Yep, it took me a minute to debug the failing test (unrelated to this PR), but now that it's all green I'll go ahead and merge. :+1:

**Comment by eddelbuettel** [comment]
Thanks so much for the persistence!   Should we make a 4.1.1 release for CRAN?

**Comment by rcurtin** [comment]
I was thinking of waiting on #3487 and other nearly-ready-to-go PRs to be merged, and then releasing the next version of mlpack (thus updating CRAN too).  So, within the next week or so?

**Comment by eddelbuettel** [comment]
That works too and is a wee bit cleaner :)

**Comment by conradsnicta** [comment]
> I was thinking of waiting on #3487 and other nearly-ready-to-go PRs to be merged, and then releasing the next version of mlpack (thus updating CRAN too). So, within the next week or so?

@rcurtin Suggest to release this as mlpack 4.2, instead of 4.1.1.  The changes are more than just bug fixes -- there are new features.  On top of that, people like new shiny things with bigger numbers.

**Comment by conradsnicta** [comment]
> I was thinking of waiting on #3487 and other nearly-ready-to-go PRs to be merged, and then releasing the next version of mlpack (thus updating CRAN too). So, within the next week or so?

@rcurtin Suggest to label the new release as mlpack 4.2, instead of 4.1.1.  The changes are more than just bug fixes -- there are new features.  On top of that, people like new shiny things with bigger numbers.

**Comment by rcurtin** [comment]
Agreed, 4.2.0 is just fine given the new features.

**Pull request merged**